### PR TITLE
Apply John Papa's Angular Code style

### LINKED ISSUE #604: Adopt John Papa Angular Styleguide?
Context: https://github.com/UW-Madison-DoIT/angularjs-portal/pull/601#discussion_r107953751

https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md
https://github.com/dustinspecker/eslint-config-angular

### PR DESCRIPTION
**Pull request by ChristianMurphy**
Apply John Papa's Angular Code style to the Angular JS portal.
Resolves #604 

:bulb: Reasoning:
It is another well known standard that can be used to improve code quality.

:notebook: Note:
This mainly intended to spark discussion.
Depending on the outcome of the #604 discussion this PR may not be of value to the project.

----

Contributor License Agreement adherence:

- [x] This Contribution is under the terms of Individual [Contributor License Agreements][] (and also Corporate Contributor License Agreements to the extent applicable) appearing in the [Apereo CLA roster][].

[Apereo CLA roster]: http://licensing.apereo.org/completed-clas
[Contributor License Agreements]: https://www.apereo.org/licensing/agreements

### CODE REVIEW

**Comment by davidmsibley** [comment]
@ChristianMurphy, I've pulled down your branch and have been working on it a bit.  It looks like things are broken mostly because of the controller-as rule.  We've been relying on the $scope inheritance hierarchy in many of the controllers, which isn't there when you replace $scope with this.

**Comment by ChristianMurphy** [comment]
Thanks for reviewing this @apetro and @davidmsibley!
@davidmsibley I am not yet familiar with where the `angularjs-portal`/`uportal-home` uses `$scope` inheritance, does the `controller-as` rule need to be completely rolled back or does it need a selective rollback?

**Comment by davidmsibley** [comment]
It's been a while since I worked with angularjs, so bare with me, I'm trying to learn as I go.  There are two things that I'm checking, the layout controller spec, and actually running the application.

In the spec, there must be some different way we need to check layout and layoutEmpty now because those variables are not getting assigned to scope anymore.

When I'm running the application, my browser is dying here: https://github.com/ChristianMurphy/angularjs-portal/blob/dfe0269ccfeb0677c714e5b4280195afca092784/angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js#L388 
That portlet variable is accessible in $scope because it's part of it's parent's parent scope, but those don't get passed down through the this object.

**Comment by davidmsibley** [comment]
figured out the problem with the tests: https://github.com/ChristianMurphy/angularjs-portal/pull/1

But the application is still breaking on run

File: angularjs-portal-home/src/main/webapp/my-app/layout/controllers.js [hunk 1]
```diff
@@ -50,15 +51,15 @@ define(['angular', 'jquery'], function(angular, $) {
        * @param nodeId
        * @param title
        */
-      this.removePortlet = function removePortletFunction(nodeId, title) {
+      vm.removePortlet = function removePortletFunction(nodeId, title) {
         layoutService.removeFromHome(nodeId, title).success(function() {
-          $scope.$apply(function(request, text) {
-            var result = $.grep($scope.layout, function(e) {
+          vm.$apply(function(request, text) {
```
Comment (davidmsibley) [inline angularjs-portal-home/src/main/webapp/my-app/layout/controllers.js]: I think this still needs to be $scope.$apply

File: angularjs-portal-home/src/main/webapp/my-app/layout/static/controllers.js [hunk 2]
```diff
@@ -1,91 +1,93 @@
...
+      vm.addToHome = function(portlet) {
         var ret = layoutService.addToHome(portlet);
         ret.success(function(request, text) {
           angular.element('.fname-' + portlet.fname)
             .html('<span style="color : green;">' +
               '<i class="fa fa-check"></i> Added Successfully</span>')
             .prop('disabled', true);
-          $scope.$apply(function() {
-            if (typeof $sessionStorage.marketplace !== 'undefined') {
+          vm.$apply(function() {
```
Comment (davidmsibley) [inline angularjs-portal-home/src/main/webapp/my-app/layout/static/controllers.js]: also here, I think this still needs to be $scope.$apply

File: angularjs-portal-home/src/main/webapp/my-app/marketplace/controllers.js [hunk 3]
```diff
@@ -57,7 +60,7 @@ define(['angular', 'jquery', 'require'], function(angular, $, require) {
             .prop('disabled', true)
             .removeClass('btn-add')
             .addClass('btn-added');
-          $scope.$apply(function() {
+          vm.$apply(function() {
```
Comment (davidmsibley) [inline angularjs-portal-home/src/main/webapp/my-app/marketplace/controllers.js]: another $scope.$apply

File: angularjs-portal-home/src/main/webapp/my-app/search/controllers.js [hunk 4]
```diff
@@ -81,88 +84,88 @@ define([
...
         // get marketplace entries
         marketplaceService.getPortlets().then(function(data) {
-            $scope.myuwResults = data.portlets;
+            vm.myuwResults = data.portlets;
             return data;
         }).catch(function() {
           $log.warn('Could not getPortlets');
         });
-        $scope.$watchGroup([
+        vm.$watchGroup([
```
Comment (davidmsibley) [inline angularjs-portal-home/src/main/webapp/my-app/search/controllers.js]: same vein, $scope.$watchGroup

**Comment by ChristianMurphy** [comment]
Thanks @davidmsibley!
I'm seeing the pattern that anything that matches `vm.$` needs to be updated.
I should be able to take a look into these this evening.

File: angularjs-portal-home/src/main/webapp/my-app/layout/controllers.js [hunk 5]
```diff
@@ -26,8 +26,10 @@ define(['angular', 'jquery'], function(angular, $) {
    * layout/partials/default-card.html)
    */
   .controller('LayoutController',
-    ['$localStorage', '$log', '$sessionStorage', '$rootScope', 'layoutService',
-    function($localStorage, $log, $sessionStorage, $rootScope, layoutService) {
+    ['$localStorage', '$log', '$sessionStorage', 'scope', '$rootScope',
```
Comment (davidmsibley) [inline angularjs-portal-home/src/main/webapp/my-app/layout/controllers.js]: this needs to be $scope

Comment (ChristianMurphy) [inline angularjs-portal-home/src/main/webapp/my-app/layout/controllers.js]: Good catch! Updated

**Comment by davidmsibley** [comment]
Status update: we're getting closer.  The main problem controllers were actually shipped off to app-framework, which doesn't necessarily solve the problem, but pushes the need for that solution to a later sprint.

I'm going through and testing the rest of the controllers now.

**Pull request closed without merging**